design/threat-model/threat-model.md

```diff
@@ -7,112 +7,113 @@ Kata Containers adds a level of isolation on top of traditional containers, the
+ and the device itself will be exposed to the guest.
+ - Control: In order to interact with the guest agent and retrieve `STDIO` from containers, a medium of communication is required.
+ This is available via `virtio-vsock`.
+ - Devices: `VFIO` is utilized when devices are passed directly to the virtual machine and exposed to the container.
+ - Dynamic Resource Management: `ACPI` is utilized to allow for dynamic VM resource management (ie, CPU, memory, device hotplug). This is required when containers are resized,
+ or more generally when containers are added to a pod. 
+ 
+How these devices are utilized varies depending on the VMM utilized. We clarify the default settings provided when integrating Kata
+with the QEMU, Firecracker and Cloud Hypervisor VMMs in the following sections.
 
-Other interfaces which may be utilized include VFIO (direct device assignment) and ACPI for dynamic VM resource management (ie, hotplug).
+### Devices
 
-How these devices are utilized varies depending on the VMM utilized. We clarify the default settings provided when integrating Kata with the QEMU VMM, Firecracker VMM, and Cloud Hypervisor VMM (integration of Cloud Hypervisor is in progress, expected in 1.10 release of Kata Containers) in the following sections.
+Each virtio device is implemented by a backend, which may execute within userspace on the host (vhost-user), the VMM itself, or within the host kernel (vhost). While it may provide enhanced performance,
+vhost devices are often seen as higher risk since an exploit would be already running within the kernel space. While VMM and vhost-user are both in userspace on the host, `vhost-user` generally allows for the back-end process to require less syscalls and capibilities compared to a full VMM.
 
-### Devices
+#### `virtio-blk and `virtio-scsi`
 
-#### virtio-blk/-scsi
+The backend for `virtio-blk` and `virtio-scsi` are based in the VMM itself (ring3 in the context of x86) by default for Cloud Hypervisor, Firecracker and QEMU.
+While `vhost` based backends are available for QEMU, it is not recommended. `vhost-user` backends are being added for Cloud Hyperivsor, they are not utilized in Kata today.
 
-The backend for virtio-blk and -scsi are based in the VMM itself (ring3 in the context of x86) by default.
-While vhost based backends are available for QEMU, it is not recommended. vhost-user backends are under consideration
-in the future, but not utilized in any of the existing Kata configurations.
+#### `virtio-fs`
 
-#### virtio-FS
+`virtio-fs` is supported in Cloud Hypervisor and QEMU. `virtio-fs`'s interaction with the host filesystem is done through a vhost-user daemon, `virtiofsd`.
+The `virtio-fs` client, running in the guest, will generate requests to access files. `virtiofsd` will receive requests, open the file, and request the VMM
+to `mmap` it into the guest. The guest will access the file using DAX, utilizing the host's page cache, thus avoiding copy and duplication.
```

Comment:
DAX isnt enabled by default today. We may want to reword this right?

Reply:
Updated - PTAL